crates/vhost/src/vhost_user/backend_req_handler.rs

```diff
@@ -519,6 +525,18 @@ impl<S: VhostUserBackendReqHandler> BackendReqHandler<S> {
                 let res = self.backend.remove_mem_region(&msg);
                 self.send_ack_message(&hdr, res)?;
             }
+            // Sets logging shared memory space.
+            // When the back-end has `VHOST_USER_PROTOCOL_F_LOG_SHMFD` protocol feature, the log
+            // memory `fd` is provided in the ancillary data of `VHOST_USER_SET_LOG_BASE` message,
+            // the size and offset of shared memory area provided in the message.
+            // See https://qemu-project.gitlab.io/qemu/interop/vhost-user.html#migration.
+            Ok(FrontendReq::SET_LOG_BASE) => {
+                self.check_proto_feature(VhostUserProtocolFeatures::LOG_SHMFD)?;
+                let file = take_single_file(files).ok_or(Error::IncorrectFds)?;
+                let msg = self.extract_request_body::<VhostUserLog>(&hdr, size, &buf)?;
+                self.backend.set_log_base(&msg, file)?;
+                self.send_reply_message(&hdr, &msg)?;
```

Comment:
According to the spec, this is wrong, because the spec says SET_LOG_BASE has no reply payload, which means it’s only expecting a reply via F_REPLY_ACK/NEED_REPLY. Whereas this (as far as I can tell) just unconditionally bounces back the request to the front end.

But as usual with vhost-user (this is a nightmare) spec and code are two different things... qemu actually does expect a response (but only with LOG_SHMFD!), but it doesn’t even care what it is. Any response, even an empty one, will do. libvhost-user also implements it this way. Honestly, I don’t even know how to fix the spec on this one. FWIW, I believe (as far as I can tell) that libvhost-user replies with a u64, so maybe we should just do the same.